NdefMessage.cpp

```diff
@@ -156,96 +200,118 @@ boolean NdefMessage::addRecord(NdefRecord& record)
     }
     else
     {
+#ifdef NDEF_USE_SERIAL
         Serial.println(F("WARNING: Too many records. Increase MAX_NDEF_RECORDS."));
+#endif
         return false;
     }
 }
 
-void NdefMessage::addMimeMediaRecord(String mimeType, String payload)
+void NdefMessage::addMimeMediaRecord(const char *mimeType, const char* payload)
```

Comment:
const char * might be technically better, but String is preferred for Arduino libraries because it's easier for beginners

Reply:
Agreed. I may have run into out of memory problems in my own applications with this library, especially on ATmega328P, and avoiding Strings was a way to mitigate this.